grovedb/src/lib.rs

```diff
@@ -274,6 +283,49 @@ impl GroveDb {
         Ok(proof_result)
     }
 
+    pub fn verify_proof(
+        path: &[&[u8]],
+        proofs: Vec<Vec<u8>>,
+        expected_root_hash: [u8; 32],
+    ) -> Result<Map, Error> {
+        // Should it really be 2 or more??
+        if proofs.len() < 2 {
+            return Err(Error::InvalidProof("Proof length should be 2 or more"));
+        }
+
+        if proofs.len() - 1 != path.len() {
+            return Err(Error::InvalidProof("Proof length should be one greater than path"));
+        }
+
+        let (mut last_root_hash, mut last_result_map) = execute_proof(&proofs[0][..]).unwrap();
+
+        for i in 1..proofs.len() {
+            if i == proofs.len() - 1 {
```

Comment:
Instead of iteration by index we can iterate by elements, also to split slice into all elements except last and a last one there is a method `split_last`